src/LanguageServer/Protocol/Handler/Diagnostics/AbstractPullDiagnosticHandler.cs

```diff
@@ -411,17 +405,11 @@ private ImmutableArray<LSP.Diagnostic> ConvertDiagnostic(IDiagnosticSource diagn
             if (capabilities.HasVisualStudioLspCapability())
             {
                 var expandedMessage = string.IsNullOrEmpty(diagnosticData.Description) ? null : diagnosticData.Description;
+                var informationService = project.Solution.Workspace.Services.GetRequiredService<IDiagnosticProjectInformationService>();
```

Comment:
Can we access this via `document.Project.Solution.Services` (to avoid the direct workspace reference in LSP handlers where possible)?

Reply:
yes.

Reply:
done.